lib/Parse/ParseDecl.cpp

```diff
@@ -1623,7 +1623,8 @@ bool Parser::MightBeDeclarator(unsigned Context) {
       // and in block scope it's probably a label. Inside a class definition,
       // this is a bit-field.
       return Context == Declarator::MemberContext ||
-             (getLangOpts().CPlusPlus && Context == Declarator::FileContext);
+             (getLangOpts().CPlusPlus && Context == Declarator::FileContext) ||
+             getLangOpts().CheckedC;
```

Comment:
>CheckedC [](start = 27, length = 8)

For posterity, it might be helpful to update the comment above to explain why there's a special-case here for Checked C.  (I assume it's for the ':' bounds expr syntax for namespace-scope variable declarations.)